### PR TITLE
Migrate to React 18

### PR DESCRIPTION
**Pull request by aryanshah701**
# Description

- Gotta move to react 18 types.

Additional changes
- Chakra upgrade
- Remove onboarding

Closes # (issue number)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] This requires a run of `yarn install`

# How Has This Been Tested?
- Copy over the api-v2/.env.development to api-v2/.env.production.
- Run yarn prod:v2 and test the app.

# Checklist:

- [x ] I have performed a self-review of my own code
- [ ] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/frontend-v2/components/Error/ErrorBoundary.tsx [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
-      onError={clientSideErrorHandler}
-    >
-      {children}
-    </ReactErrorBoundary>
+    // <ReactErrorBoundary
+    //   FallbackComponent={ClientSideError}
+    //   onError={clientSideErrorHandler}
+    // >
+    <div>{children}</div>
+    // </ReactErrorBoundary>
```
Comment (aryanshah701) [inline packages/frontend-v2/components/Error/ErrorBoundary.tsx:20]: error boundary library is incompatible with react 18, will switch out for a compatible one and style it in a separate PR

File: packages/frontend-v2/package.json [hunk 2]
```diff
@@ -4,23 +4,24 @@
...
     "@fontsource/montserrat-alternates": "^4.5.9",
     "@graduate/common": "workspace:*",
     "framer-motion": "^6",
     "immer": "^9.0.15",
-    "next": "12.1.4",
-    "react": "18.0.0",
-    "react-dom": "18.0.0",
+    "next": "latest",
+    "react": "latest",
+    "react-dom": "latest",
```
Comment (AlpacaFur) [inline packages/frontend-v2/package.json:24]: Correct me if I'm wrong, but I think we might still want concrete version numbers here so we don't encounter surprise issues when React or Next get a major version update.

Comment (aryanshah701) [reply packages/frontend-v2/package.json:24]: definitely!! thank you for this!

**Review by rael346 — APPROVED** [review]
LGTM!

**Pull request merged**